openpype/tools/launcher/models.py

```diff
@@ -222,6 +241,63 @@ def filter_compatible_actions(self, actions):
             key=lambda action: (action.order, action.name)
         )
 
+    def update_force_not_open_workfile_settings(self, is_checked, action):
+        """Store/remove config for forcing to skip opening last workfile.
+
+        Args:
+            is_checked (bool): True to add, False to remove
+            action (ApplicationAction)
+        """
+        actual_data = self._prepare_compare_data(action)
+
+        stored = self.launcher_registry.get_item("force_not_open_workfile")
+        if is_checked:
+            stored.append(actual_data)
+        else:
+            final_values = []
+            for config in stored:
+                if config != actual_data:
+                    final_values.append(config)
+            stored = final_values
+
+        self.launcher_registry.set_item("force_not_open_workfile", stored)
+
+    def is_force_not_open_workfile(self, item, stored):
+        """Checks if application for task is marked to not open workfile
+
+        There might be specific tasks where is unwanted to open workfile right
+        always (broken file, low performance). This allows artist to mark to
+        skip opening for combination (project, asset, task_name, app)
+
+        Args:
+            item (QStandardItem)
+            stored (list) of dict
+        """
+        action = item.data(ACTION_ROLE)
+        if isinstance(action, list) and action:
+            action = action[0]
+
+        if ApplicationAction not in action.__bases__:
+            return False
+
+        actual_data = self._prepare_compare_data(action)
+        for config in stored:
+            if config == actual_data:
+                return True
+
+        return False
+
+    def _prepare_compare_data(self, action):
+        if isinstance(action, list) and action:
```

Comment:
Just a note. If `action` is empty it will crash on return because is calling `action.label.lower()`.

Reply:
Fixed.